### PR TITLE
scylla_bench_thread: make it support docker image with no shell

### PR DESCRIPTION
**Pull request by dkropachev**
Closes https://github.com/scylladb/scylla-cluster-tests/issues/8721

### Testing

- [ ]

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Comment by dkropachev** [comment]
@vponomaryov, can you please take a look if it is good enought

File: sdcm/scylla_bench_thread.py [hunk 1]
```diff
@@ -203,6 +207,8 @@ def _run_stress(self, loader, loader_idx, cpu_idx):  # pylint: disable=too-many-
             )
             cmd_runner_name = loader.ip_address
 
+        is_shell_supported = cmd_runner.run(cmd="sh -c true", timeout=self.timeout, retry=0).returncode == 0
```
Comment (vponomaryov) [inline sdcm/scylla_bench_thread.py:210]: Not being 0 it will raise exception.
Need to use `ignore_status=True` in such a case.

File: sdcm/scylla_bench_thread.py [hunk 2]
```diff
@@ -183,7 +185,9 @@ def create_stress_cmd(self, stress_cmd, loader, cmd_runner):
                     # server_names = ' '.join(f'-tls-server-name {ip}' for ip in ips.split(","))
                     # stress_cmd = f'{stress_cmd.strip()} -tls-host-verification {server_names}'
 
-        return stress_cmd
+        if is_shell_supported:
+            return stress_cmd
+        return stress_cmd.removeprefix('scylla-bench')
```
Comment (vponomaryov) [inline sdcm/scylla_bench_thread.py:190]: There is logic that depends on the tool/binary name that is going to be used.
For example, dynamic loader in K8S uses it.
Maybe somewhere else.

File: sdcm/utils/docker_remote.py [hunk 3]
```diff
@@ -102,7 +106,9 @@ def get_log(self):
         return self.node.remoter.run(f"{self.sudo_needed} docker logs {self.docker_id}").stdout.strip()
 
     def run(self, cmd, *args, **kwargs):
-        return self.node.remoter.run(f'{self.sudo_needed} docker exec {self.docker_id} /bin/sh -c {shlex.quote(cmd)}', *args, **kwargs)
+        if self.is_shell_supported:
+            return self.node.remoter.run(f'{self.sudo_needed} docker exec {self.docker_id} /bin/sh -c {shlex.quote(cmd)}', *args, **kwargs)
+        return self.node.remoter.run(f'{self.sudo_needed} docker exec {self.docker_id} {cmd}', *args, **kwargs)
```
Comment (vponomaryov) [inline sdcm/utils/docker_remote.py:111]: I think that something in SCT may fail without `shlex.quote` usage in this case.

**Review by vponomaryov — COMMENTED** [review]
Comments inline. But in general I am not sure it is the way we should follow.

This implicit approach looks fragile and we have a lot of different places where we use either shell env vars or pipes...
So, personally, I would prefer to have shell in each loader image.

@fruch and @soyacz WDYT?

File: sdcm/scylla_bench_thread.py [hunk 4]
```diff
@@ -183,7 +184,9 @@ def create_stress_cmd(self, stress_cmd, loader, cmd_runner):
...
```
Comment (fruch) [inline sdcm/scylla_bench_thread.py:189]: this change is not related to have sh or not.
it's depend on having an entrypoint set to what.

I rather align this to set explicitly the entry point SCT expect, then modifying the command, based on unrelated logic

File: sdcm/scylla_bench_thread.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/scylla_bench_thread.py]: > Not being 0 it will raise exception. Need to use `ignore_status=True` in such a case.

fixed.

Comment (dkropachev) [reply sdcm/scylla_bench_thread.py]: > this is not an assert statement...

?

File: sdcm/scylla_bench_thread.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/scylla_bench_thread.py:189]: Fixed that part of logic to use env instead

File: sdcm/utils/docker_remote.py (hunk 3 above)
Comment (dkropachev) [reply sdcm/utils/docker_remote.py]: fixed

**Comment by fruch** [comment]
@dkropachev do we still need this one ?

**Comment by dkropachev** [comment]
this PR is getting too intrusive, closing it, will make a special dockerfile with a shell in it.
